lib/trace-writer.js

```diff
@@ -44,6 +45,12 @@ function TraceWriter(logger, config) {
   /** @private {Array<string>} stringified traces to be published */
   this.buffer_ = [];
 
+  if (config.samplingRate === -1) {
```

Comment:
This code belongs in tracing-policy.js. Trace-writer should not be in the business of looking at config.samplingRate.